src/iperf_util.c

```diff
@@ -433,56 +434,65 @@ iperf_dump_fdset(FILE *fp, char *str, int nfds, fd_set *fds)
     fprintf(fp, "]\n");
 }
 
-/* daemon(3) implementation for systems lacking one */
+/*
+ * daemon(3) implementation for systems lacking one.
+ * Cobbled together from various daemon(3) implementations,
+ * not intended to be general-purpose. */
 #ifndef HAVE_DAEMON
 int daemon(int nochdir, int noclose)
 {
     pid_t pid = 0;
     pid_t sid = 0;
+    int fd;
+
+    /*
+     * Ignore any possible SIGHUP when the parent process exits.
+     * Note that the iperf3 server process will eventually install
+     * its own signal handler for SIGHUP, so we can be a little
+     * sloppy about not restoring the prior value.  This does not
+     * generalize.
+     */
+    signal(SIGHUP, SIG_IGN);
 
     pid = fork();
     if (pid < 0) {
 	    return -1;
     }
-
     if (pid > 0) {
-	exit(0);
+	/* Use _exit() to avoid doing atexit() stuff. */
+	_exit(0);
     }
 
-    umask(0);
-
     sid = setsid();
     if (sid < 0) {
 	return -1;
     }
 
-    signal(SIGHUP, SIG_IGN);
+    /*
+     * Fork again to avoid becoming a session leader.
+     * This might only matter on old SVr4-derived OSs. 
```

Comment:
Anyone still running SVR4 should be led through the streets with people shouting "Shame!" at them, as they clearly don't care about being a beachhead for botnet attacks, etc...

Reply:
SVr4-*derived* OSs.  I don't mind getting rid of this except that the person who originally filed a bug was running IRIX, which is very much based on SVr4.  So I don't want to go through all this trouble just to have something that won't work for them.

Reply:
(Whether we should be expending this amount of effort on such old platforms is a different question, but I figure we've put this much effort into it, we may as well finish it off.)

Reply:
> So I don't want to go through all this trouble just to have something that won't work for them.

Right, I just thought you said something about iperf3 installing its own handler anyway.  In any case, nothing major...

Reply:
OK.  I really appreciate your feedback and discussion.  At this point I think I want to merge this code to get it out into the world for more testing (I've done about all I call given that I don't have a platform that would actually use this daemon() implementation, so I need to fake out some stuff just to test it).  If something doesn't work, presumably somebody will yell about it and I'll fix it.  Thanks again!